flycheck.el

```diff
@@ -4297,17 +4297,23 @@ POS defaults to `point'."
 
 (defun flycheck-jump-to-error (error)
   "Go to the location of ERROR."
-  (let ((error-copy (copy-flycheck-error error))
-        (buffer (find-file-noselect (flycheck-error-filename error))))
-    (setf (flycheck-error-buffer error-copy) buffer)
-    (flycheck-jump-in-buffer buffer error-copy)
-    ;; When jumping to an error in another file, it may not have
-    ;; this error available for highlighting yet, so we trigger a check
-    ;; if necessary.
-    (with-current-buffer buffer
-      (unless (seq-contains flycheck-current-errors error-copy 'equal)
-        (when flycheck-mode
-          (flycheck-buffer))))))
+  (let* ((error-copy (copy-flycheck-error error))
+         (filename (flycheck-error-filename error))
+         (other-file-error (flycheck-relevant-error-other-file-p error))
+         (buffer (if filename
+                     (find-file-noselect filename)
+                   (flycheck-error-buffer error))))
+    (when (buffer-live-p buffer)
+      (setf (flycheck-error-buffer error-copy) buffer)
+      (flycheck-jump-in-buffer buffer error-copy)
+      ;; When jumping to an error in another file, it may not have
+      ;; this error available for highlighting yet, so we trigger a check
+      ;; if necessary.
+      (when other-file-error
+        (with-current-buffer buffer
+          (unless (seq-contains flycheck-current-errors error-copy 'equal)
+            (when flycheck-mode
+              (flycheck-buffer))))))))
```

Comment:
Looks great now.  We could move the error copy down into the "unless", but it's a minor point.

Reply:
The point of the error copy is for the `flycheck-jump-in-buffer` call above.  We would have to duplicate that path as well?

Reply:
No, I think we could just pass the original error to `flycheck-jump-in-buffer`.

Reply:
Sorry, I was misreading `flycheck-jump-in-buffer`. I got confused by the fact that we pass both a buffer and an error to that function, and it uses both the buffer we pass it *and* the buffer in the error (and presumably expects them to be the same?)

Reply:
> it uses both the buffer we pass it and the buffer in the error (and presumably expects them to be the same?)

Yes.  Now that you mention it, we could probably refactor that.  The point is that `flycheck-error-pos` needs the error struct to have the right buffer, or the position will be wrong.

Reply:
Yep, it's very clear now.  Does it make sense to call that function with another buffer than the one in the error? If not, we could drop that extra argument.
 
But let's not delay this independent fix too much, either.  Your patch is quite fine :) You could just push it with a FIXME.